protocol/encryption/protocol_message.proto

```diff
@@ -54,13 +54,24 @@ message X3DHHeader {
   bytes id = 4;
 }
 
+// Hash Ratchet Header
+message HRHeader {
+  // community key ID
+  uint32 key_id = 1;
+  // Community message number for this key_id
+  uint32 seq_no = 2;
+  // Community ID
+  string group_id = 3;
+}
+
 // Direct message value
-message DirectMessageProtocol {
+message EncryptedMessageProtocol {
   X3DHHeader X3DH_header = 1;
-  DRHeader DR_header = 2;
+  DRHeader DR_header = 2; 
+  HRHeader HR_header = 3;
   DHHeader DH_header = 101;
   // Encrypted payload
-  bytes payload = 3;
+  bytes payload = 4;
```

Comment:
mmh, this looks wrong, it can't be changed,  for backward compatibility, you will have to go out of order with the numbering :)